CMakeLists.txt

```diff
@@ -11,16 +11,37 @@ add_compile_options ( -Werror
 
 ################################################################################
 # Set up Sparta
-if (NOT SPARTA_BASE)
-  message (FATAL_ERROR "Need to define the location of the Sparta: cmake <args> -DSPARTA_BASE=\"/path/to/sparta\"")
-endif ()
-include (${SPARTA_BASE}/cmake/sparta-config.cmake)
+if(IS_DIRECTORY ${SPARTA_SEARCH_DIR})
+  message (STATUS "Using '${SPARTA_SEARCH_DIR}' for sparta install")
+elseif(DEFINED ENV{CONDA_PREFIX})
+  message (STATUS "Looking for SPARTA in the conda environment: '$ENV{CONDA_PREFIX}'")
+  set(SPARTA_SEARCH_DIR $ENV{CONDA_PREFIX})
+else()
+  message (STATUS "If needed, please provide the location where sparta is installed: -DSPARTA_SEARCH_DIR=<directory>")
+endif()
+set(CMAKE_MODULE_PATH "${SPARTA_SEARCH_DIR}/lib/cmake/sparta" ${CMAKE_MODULE_PATH})
+
+find_package(Sparta REQUIRED)
+
+if (NOT SPARTA_FOUND)
+  message (FATAL_ERROR "Sparta was not found in ${SPARTA_SEARCH_DIR}")
+else()
+  message (STATUS "Sparta was found in ${SPARTA_SEARCH_DIR}")
+endif()
+#include_directories(${SPARTA_INCLUDE_DIRS})
```

Comment:
I'm wondering if uncommenting this line with fix your include issues.  Can you do that AND remove the target_link lines to see if you still have problems?

Reply:
It's what this line and the next in `FindSparta.cmake` are providing automatically when you add sparta as a link library:

https://github.com/sparcians/map/blob/6f0636f7a717238cd039be33f7975048449bb82c/sparta/cmake/FindSparta.cmake#L69

Reply:
Ok, I can reproduce the missing include problem, but the not boost one.  I'll admit that I don't fully grok how `cmake` brings in the goods from a package.  You'd think the `find_package` method would be enough!  But sparta must be installed in a "standard" location or something so that includes AND library paths are set up?  I don't get it...

Reply:
All `find_package` does is import a target for the library (as if it were defined elsewhere in the makefiles with `add_library`) but it won’t set up any default compile or link flags project wide, it will only add the relevant options to the targets when you explicitly say they depend on the imported target. In this case that means unless you explicitly add SPARTA::sparta it won’t pass the flags. Which you won’t notice if everything is installed in system wide include and lib paths, but it will break as soon as one of the components (either Sparta or the lobs it depends on) is not in that default path. 

That said, there could be a way to only add the include paths and maybe not the link flags, but I expect that to fail. 

Reason I see boost problems is because I have multiple boost versions via homebrew, and Sparta needs one that is not the default one. (There’s macOS issues with boost 1.81 command line arguments)

Reply:
Thank you Peter and Knute for working through this. 

Trying to understand the latest status of this work: From the comments above, I understand at least the regression is not working yet. But is it just the regression or more than that? Would I be able to build riscv-perf-model / olympia with latest head of master of Sparta (which installs Sparta in the conda environment)?

Reply:
As long as Sparta (and it’s dependencies like boost hdf5 etc) are available in the default paths (line and includes) it just works. It’s just in case some dependencies are not in a default location , that the overridden paths are not passed through from Sparta to Olympia at this point… (eg I have a home brew boost version in macOS that is not the default one which gives me troubles when I want to this locally branch currently)